src/main.c

```diff
@@ -81,6 +82,10 @@ static int hd_clutter_mutex_enabled = FALSE;
 static int hd_clutter_mutex_do_unlock_after_disabling = FALSE;
 static GMutex hd_clutter_mutex;
 
+static int xi_motion_ev_type = -1;
+static int xi_presence_ev_type = -1;
+static gboolean xdevice_is_ts[64];
```

Comment:
I would just use GArray here. Having 256 bytes of static memory occupied for 8 bytes (at most) significant data is just an overkill. GArray in turn allows you to dynamically resize and quickly access items.

Reply:
ok, makes sense, will rework it to use GArray